torch/utils/data/_utils/collate.py

```diff
@@ -26,11 +26,21 @@ def default_convert(data):
             return data
         return torch.as_tensor(data)
     elif isinstance(data, collections.abc.Mapping):
-        return {key: default_convert(data[key]) for key in data}
+        try:
+            return elem_type({key: default_convert(data[key]) for key in data})
+        except TypeError:
+            # The mapping type may not support `__init__(iterable)`.
+            return {key: default_convert(data[key]) for key in data}
     elif isinstance(data, tuple) and hasattr(data, '_fields'):  # namedtuple
         return elem_type(*(default_convert(d) for d in data))
+    elif elem_type is tuple:
+        return [default_convert(d) for d in data]  # Backwards compatibility.
```

Comment:
Could you please use `issubclass(elem_type, tuple)`?

Reply:
So we wanna use lists for tuples and their subclasses?

Reply:
Yes